_posts/2024-10-16-18.0.0-release.md

```diff
@@ -0,0 +1,242 @@
+
+### GPU
+
+
+### IPC
+
+The seed corpus used for fuzzing the IPC reader has been improved, hopefully
+helping make the IPC reader even more robust against corrupt or malicious
+IPC streams (GH-38041).
+
+### Parquet
+
+A new command line utility `parquet-dump-footer` allows dumping the Thrift-encoded
+footer metadata of a Parquet file, optionally scrubbing confidential data
+(GH-42102). This is part of the effort to collect real-world Parquet metadata
+so as to evaluate the efficiency of future improvements to the Parquet format.
+Please see https://github.com/apache/parquet-benchmark for instructions to submit
+footers representative of your own workloads.
+
+### Substrait
+
+
+## C# notes
+
+- Partial support has been added for LargeBinary, LargeString and LargeList. The column sizes cannot exceed 2 GB in length. (GH-43266).
+- Changes to Flight support were made for better control and compatibility, and to allow Flight Server to be hosted in pre-Kestrel versions of .NET (GH-43907, GH-43672, GH-41347).
+- Support has been added for newly-defined types decimal32 and decimal64 (GH-44271).
+- The import of sliced arrays through the C Data interface now works correctly. (GH-43267)
+## Java notes
+
+**Java 8 is no longer supported.** (GH-38051)
+
+**Gandiva may not work in this release.** For details, please see [GH-43576](https://github.com/apache/arrow/issues/43576).
+
+Basic support for RunEndEncoded was added (GH-39982). The ListView/StringView vector implementations are now more complete, including C Data support (multiple issues).
+
+Several APIs have been updated to accept `long` for addresses in preparation for FFM/large buffer support (GH-43902). We no longer expose `sun.misc.Unsafe` (GH-43479). We no longer ship the `shaded` flight-core JARs (GH-43217).
+
+More options were added to the Dataset ScanOptions API (GH-28866).
+
+## JavaScript notes
+
+
+## Python notes
+Compatibility notes:
+* NumPy required dependency has been removed from pyarrow packaging
+  [GH-43846](https://github.com/apache/arrow/issues/43846) and has been
+  made an optional runtime dependency [GH-25118](https://github.com/apache/arrow/issues/25118).
+* Support for Python 3.8 has been dropped [GH-43518](https://github.com/apache/arrow/issues/43518)
+* No longer used serialize/deserialize Pyarrow C++ functions have been
+  deprecated [GH-44063](https://github.com/apache/arrow/issues/44063).
+* Passing of build flags to setup.py (e.g. `setup.py --with-parquet`) has been
+  deprecated [GH-43514](https://github.com/apache/arrow/issues/43514)
+
+New features:
+* Non-cpu work has continued with [GH-43973](https://github.com/apache/arrow/issues/43973),
+  [GH-43728](https://github.com/apache/arrow/issues/43728), [GH-43727](https://github.com/apache/arrow/issues/43727),
+  [GH-43391](https://github.com/apache/arrow/issues/43391),
+  [GH-42222](https://github.com/apache/arrow/issues/42222) and
+  [GH-41665](https://github.com/apache/arrow/issues/41665).
```

Comment:
I think just enumerating those issue links might not be that informative without more context (for someone reading this, you would have to click on all those links to have an idea what actually changed)